marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadUserArtifactsCommand.java

```diff
@@ -362,23 +290,65 @@ private void loadSteps(DatabaseClient stagingClient) throws IOException {
-            for (File stepTypeDir : stepsPath.toFile().listFiles(file -> file.isDirectory())) {
+            for (File stepTypeDir : stepsPath.toFile().listFiles(File::isDirectory)) {
                 final String stepType = stepTypeDir.getName();
                 for (File stepFile : stepTypeDir.listFiles((File d, String name) -> name.endsWith(".step.json"))) {
                     JsonNode step = objectMapper.readTree(stepFile);
                     if (!step.has("name")) {
                         throw new RuntimeException("Unable to load step from file: " + stepFile + "; no 'name' property found");
                     }
                     final String stepName = step.get("name").asText();
-                    if (logger.isInfoEnabled()) {
-                        logger.info(format("Loading step of type '%s' with name '%s'", stepType, stepName));
-                    }
+                    logger.info(format("Loading step of type '%s' with name '%s'", stepType, stepName));
                     stepService.saveStep(stepType, step);
                 }
             }
         }
     }
 
+    private void loadFlows(DatabaseClient stagingClient) throws IOException {
+        final Path flowsPath = hubConfig.getHubProject().getFlowsDir();
+        if (flowsPath.toFile().exists()) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            ArtifactService service = ArtifactService.on(stagingClient);
+            for (File file : flowsPath.toFile().listFiles(f -> f.isFile() && f.getName().endsWith(".flow.json"))) {
+                JsonNode flow = objectMapper.readTree(file);
+                if (!flow.has("name")) {
+                    throw new RuntimeException("Unable to load flow from file: " + file + "; no 'name' property found");
+                }
+                final String flowName = flow.get("name").asText();
+                logger.info(format("Loading flow with name '%s'", flowName));
+                service.setArtifact("flow", flowName, flow);
+            }
+        }
+    }
+
+    private void loadStepDefinitions(DatabaseClient stagingClient) throws IOException {
+        final Path stepDefsPath = hubConfig.getHubProject().getStepDefinitionsDir();
+        if (stepDefsPath.toFile().exists()) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            ArtifactService service = ArtifactService.on(stagingClient);
+            for (File typeDir : stepDefsPath.toFile().listFiles(File::isDirectory)) {
+                final String stepDefType = typeDir.getName();
+                for (File defDir : typeDir.listFiles(File::isDirectory)) {
+                    final String stepDefName = defDir.getName();
+                    File stepDefFile = new File(defDir, stepDefName + ".step.json");
+                    if (stepDefFile.exists()) {
+                        JsonNode stepDef = objectMapper.readTree(stepDefFile);
+                        if (!stepDef.has("name")) {
```

Comment:
Same as previous comment. /data-hub/5/artifacts/stepDefinition.sjs has 'name' as a requiredProperty. Do we need this additional validation in the client ?

Reply:
Same thing here too, we need it for the second parameter.